app/controllers/AddAdditionalMetadataController.scala

```diff
@@ -125,14 +126,24 @@ class AddAdditionalMetadataController @Inject() (
     }
   }
 
-  private def getFormFields(consignmentId: UUID, request: Request[AnyContent]): Future[List[FormField]] = {
+  private def getFormFields(consignmentId: UUID, request: Request[AnyContent], metadataType: String): Future[List[FormField]] = {
+    val closure: Boolean = metadataType == "closure"
     for {
+      displayProperties <-
+        if (closure) { Future(List()) }
+        else {
+          displayPropertiesService.getDisplayProperties(consignmentId, request.token.bearerAccessToken)
+        }
       customMetadata <- customMetadataService.getCustomMetadata(consignmentId, request.token.bearerAccessToken)
-      customMetadataUtils = new CustomMetadataUtils(customMetadata)
-
-      dependencyProperties: Set[CustomMetadata] = getDependenciesForValue(customMetadataUtils, closureType.name, closureType.value)
-
-      formFields = customMetadataUtils.convertPropertiesToFormFields(dependencyProperties)
+      formFields =
+        if (closure) {
+          val customMetadataUtils = new CustomMetadataUtils(customMetadata)
+          val dependencyProperties: Set[CustomMetadata] = getDependenciesForValue(customMetadataUtils, closureType.name, closureType.value)
+          customMetadataUtils.convertPropertiesToFormFields(dependencyProperties)
+        } else {
+          val descriptiveProperties = displayProperties.filter(dp => dp.active && dp.propertyType.toLowerCase == metadataType)
```

Comment:
Can we move this filter part in the service so that we can directly assign the properties in the `displayProperties` variable at line number 133 and we can delete the if-else part at line number 133?

Reply:
Yeah, that makes sense. Thanks

Reply:
I think we forgot to remove the filter at line number 140. It's not required now.

`displayProperties.filter(dp => dp.active && dp.propertyType.toLowerCase == metadataType)`